### ISSUE TITLE
Don't overwrite stake in addCandidate

### ISSUE THREAD

**Issue by rbharath** [computablelabs/goest#137]
At present, `addCandidate` in `Voting.vy` will overwrite the owner's stake. If a address has submitted a candidate with the same hash twice (say one challenge submitted, resolved, then resubmitted with the same hash), then it's possible to destroy some stake. This happens in the following line (https://github.com/computablelabs/goest/blob/master/contracts/Voting.vy#L136):
```
self.stakes[owner][hash] = stake
```
To fix this, we should swap to accumulating the stake with
```
self.stakes[owner][hash] += stake
```

**Comment by rbharath** [comment]
Fixed by #145.

**Comment by rbharath** [comment]
Re-opening after rollback

**Comment by rbharath** [comment]
Implemented by #164
